### PR TITLE
Update GitHub workflows

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
- Fix starting minishift cluster by adding `--registry-mirror=https://quay.io`
- Merge `update-crd-files.sh` and `update-nightly-bundle.sh` into one single script `update-resources.sh`
- Introduce `check-resources.sh` corresponding to `update-resources.sh`
- Removes resources updating from the release script, added validation (if resources aren't up to date then a separate PR has to be created)

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by flacatus — APPROVED** [review]

File: .github/bin/check-resources.sh [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    pushd "${ROOT_PROJECT_DIR}"
+    source "${ROOT_PROJECT_DIR}/olm/update-resources.sh"
+
+    changedFiles=($(git diff --name-only))
+
+    # Check if there are any difference in the crds. If yes, then fail check.
+    if [[ " ${changedFiles[*]} " =~ $CRD_V1 ]] || [[ " ${changedFiles[*]} " =~ $CRD_V1BETA1 ]]; then
+        echo "[ERROR] CRD file is not up to date: ${BASH_REMATCH}"
+        echo "[ERROR] Run 'olm/update-resources.sh' to regenrated CRD files."
```
Comment (AndrienkoAleksandr) [inline .github/bin/check-resources.sh:67]:
```suggestion
        echo "[ERROR] Run 'olm/update-resources.sh' to regenerate CRD files."
```

File: .github/bin/check-resources.sh [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  local CRD_FILE_KUBERNETES="deploy/olm-catalog/nightly/eclipse-che-preview-kubernetes/manifests/org_v1_che_crd.yaml"
+  local CRD_FILE_OPENSHIFT="deploy/olm-catalog/nightly/eclipse-che-preview-openshift/manifests/org_v1_che_crd.yaml"
+
+  pushd "${ROOT_PROJECT_DIR}" || true
+
+  changedFiles=($(git diff --name-only))
+  if [[ " ${changedFiles[*]} " =~ $CSV_FILE_OPENSHIFT ]] || [[ " ${changedFiles[*]} " =~ $CSV_FILE_OPENSHIFT ]] || \
+     [[ " ${changedFiles[*]} " =~ $CRD_FILE_KUBERNETES ]] || [[ " ${changedFiles[*]} " =~ $CRD_FILE_OPENSHIFT ]]; then
+    echo "[ERROR] Nighlty bundle is not up to date: ${BASH_REMATCH}"
+    echo "[ERROR] Run 'olm/update-resources.sh' to regenrated CSV/CRD files."
```
Comment (AndrienkoAleksandr) [inline .github/bin/check-resources.sh:88]:
```suggestion
    echo "[ERROR] Run 'olm/update-resources.sh' to regenerate CSV/CRD files."
```

File: README.md [hunk 3]
```diff
@@ -331,16 +331,10 @@ New golang dependencies in the vendor folder should be committed and included in
 
 ### Updating Custom Resource Definition file
 
-Che cluster custom resource definition (CRD) defines Eclipse CheCluster custom resource object. It contains information about object structure, field types, field descriptions.
-CRD file is a YAML definition located in the `deploy/crds/org_v1_che_crd.yaml`. The file is auto-generated, so do not edit it directly to update it. If you want to add new fields or fix descriptions in the CRD, make your changes in the file `pkg/apis/org/v1/che_types.go` and run VSCode task `Update crd files` or use the terminal
+Che cluster custom resource definition (CRD) defines Eclipse CheCluster custom resource object. It contains information about object structure, field types, field descriptions. CRD file is a YAML definition located in the `deploy/crds/org_v1_che_crd.yaml`. The file is auto-generated, so do not edit it directly to update it. If you want to add new fields or fix descriptions in the CRD, make your changes in the file `pkg/apis/org/v1/che_types.go` and run VSCode task `Update resources` or use the terminal
```
Comment (AndrienkoAleksandr) [inline README.md:334]:
```suggestion
Che cluster custom resource definition (CRD) defines Eclipse CheCluster custom resource object. It contains information about object structure, field types, field descriptions. CRD file is a YAML definition located in the folder `deploy/crds`. These files are auto-generated, so do not edit it directly to update them. If you want to add new fields or fix descriptions in the CRDs, make your changes in the file `pkg/apis/org/v1/che_types.go` and run VSCode task `Update resources` or use the terminal
```

File: olm/update-resources.sh [hunk 4]
```diff
@@ -0,0 +1,254 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Red Hat, Inc.
```
Comment (AndrienkoAleksandr) [inline olm/update-resources.sh:3]: 2021

File: .github/workflows/pr-main-check.yml [hunk 5]
```diff
@@ -0,0 +1,94 @@
+#
+#  Copyright (c) 2012-2020 Red Hat, Inc.
```
Comment (AndrienkoAleksandr) [inline .github/workflows/pr-main-check.yml:2]: Minor 2021

File: .github/bin/check-resources.sh [hunk 6]
```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+#
+# Copyright (c) 2012-2020 Red Hat, Inc.
```
Comment (AndrienkoAleksandr) [inline .github/bin/check-resources.sh:3]: Minor 2021

**Review by AndrienkoAleksandr — APPROVED** [review]

**Pull request merged**